server/sonar-server/src/main/java/org/sonar/server/permission/ws/GroupsAction.java

```diff
@@ -83,58 +78,42 @@ public void define(WebService.NewController context) {
       .setResponseExample(Resources.getResource(getClass(), "groups-example.json"))
       .setHandler(this);
 
-    createPermissionParameter(action)
-      .setRequired(false);
+    // TODO add orgKey
+    createPermissionParameter(action).setRequired(false);
     createProjectParameters(action);
   }
 
   @Override
-  public void handle(Request wsRequest, Response wsResponse) throws Exception {
-    GroupsWsRequest groupsRequest = toGroupsWsRequest(wsRequest);
-    WsGroupsResponse groupsResponse = doHandle(groupsRequest);
-    writeProtobuf(groupsResponse, wsRequest, wsResponse);
-  }
-
-  private WsGroupsResponse doHandle(GroupsWsRequest request) {
+  public void handle(Request request, Response response) throws Exception {
     DbSession dbSession = dbClient.openSession(false);
     try {
-      Optional<ComponentDto> project = dependenciesFinder.searchProject(dbSession, newOptionalWsProjectRef(request.getProjectId(), request.getProjectKey()));
-      checkProjectAdminUserByComponentDto(userSession, project);
-
-      PermissionQuery dbQuery = buildPermissionQuery(request, project);
-      List<GroupDto> groups = findGroups(dbSession, dbQuery);
-      int total = dbClient.groupPermissionDao().countGroupsByPermissionQuery(dbSession, dbQuery);
-      List<GroupPermissionDto> groupsWithPermission = findGroupPermissions(dbSession, groups, project);
-      return buildResponse(groups, groupsWithPermission, Paging.forPageIndex(request.getPage()).withPageSize(request.getPageSize()).andTotal(total));
+      Optional<ProjectId> projectId = support.findProject(dbSession, request);
+      checkProjectAdminUserByComponentUuid(userSession, projectId.isPresent() ? projectId.get().getUuid() : null);
+
+      PermissionQuery query = buildPermissionQuery(request, projectId);
+      // TODO validatePermission(groupsRequest.getPermission(), wsProjectRef);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)